Develop

alenaa-bva opened the pull request.

Commit:
добавлена структура проекта

alenaa-bva committed:
Добавлен первый тест по вопросам и ответам

Commit:
Стабильно выполняющийся тест test_get_answer_by_click_on_question

alenaa-bva committed:
Добавлены тесты и отчет allure

klushcheva commented on data.py:
```diff
@@ -0,0 +1,25 @@
+class BasePageData:
+    home_page_url = "https://qa-scooter.praktikum-services.ru/"
```
Можно лучше: константы обозначаем капслоком https://peps.python.org/pep-0008/#constants

klushcheva commented on pages/scooter_base_page.py:
```diff
@@ -0,0 +1,18 @@
+from helpers import wait_element_to_be_clickable, scroll_to_element
+from locators import BasePageLocators
+
+class ScooterBasePage:
```
Нужно поправить: мы используем паттерн PageObject: значит, должен быть BasePage,  от которого **наследуются** остальные страницы
В BasePage должны быть основные методы, такие, как: подождать и кликнуть на элемент, перейти по url и так далее

klushcheva commented on pages/scooter_home_page.py:
```diff
@@ -0,0 +1,76 @@
+    def __init__(self, driver):
+        self.driver = driver
+
+    @staticmethod
```
Поправить: методы не должны быть статичными

klushcheva commented on tests.py:
```diff
@@ -0,0 +1,24 @@
+base_url = 'https://qa-mesto.praktikum-services.ru/'
+
+def test_check_email_placeholder():
+    driver = webdriver.Chrome()
```
Поправить: Инициализация и закрытие драйвера - только через фикстуру

klushcheva commented on tests/test_home_page.py:
```diff
@@ -0,0 +1,69 @@
+                        )
+    def test_get_answer_by_click_on_question(self, driver):
+
+        expected_answers = HomePageData.answers_data
```
Нужно поправить: тексты вопросов также должны проверяться

alenaa-bva committed:
1. Методы из helpers перенесены в scooter_base_page.py
2. ScooterBasePage родительский класс для классов ScooterHomePage и ScooterOrderPage
3. Добавлена проверка вопросов в тесте tests.test_order_page.TestOrderPage.test_create_an_order
4. Все методы не статические

alenaa-bva committed:
Удален лишний файл tests.py
